src/core/public/http/abortable.ts

```diff
@@ -17,23 +17,22 @@
  * under the License.
  */
 
-jest.mock('../chrome', () => ({
-  addBasePath: (path: string) => `http://localhost/myBase/${path}`,
-}));
+import { HttpHandler, HttpFetchOptions, AbortablePromise } from './types';
 
-jest.mock('../metadata', () => ({
-  metadata: {
-    version: 'my-version',
-  },
-}));
+export function abortable<T>(handler: HttpHandler<T>) {
+  return (path: string, options: HttpFetchOptions = {}): AbortablePromise<T> => {
+    const controller = options.signal
+      ? { signal: options.signal, abort: Function.prototype }
+      : new AbortController();
+    const promise = handler(path, { ...options, signal: controller.signal });
 
-import { kfetchAbortable } from './kfetch_abortable';
-
-describe('kfetchAbortable', () => {
-  it('should return an object with a fetching promise and an abort callback', () => {
-    const { fetching, abort } = kfetchAbortable({ pathname: 'my/path' });
-    expect(typeof fetching.then).toBe('function');
-    expect(typeof fetching.catch).toBe('function');
-    expect(typeof abort).toBe('function');
-  });
-});
+    // NOTE: We are only using Object.defineProperty with enumerable:false to be explicit about
+    // excluding promise.abort() which isn't the case with Object.assign.
+    return Object.defineProperty(promise, 'abort', {
```

Comment:
I realize I'm a little late to be suggesting changes like this, but we did this in [elasticsearch-js client](https://github.com/elastic/elasticsearch-js-legacy/blob/16.x/src/lib/transport.js#L178-L180) and I personally regretted it after having to explain to users how `await` was causing `undefined is not a function` type errors.

Have we considered just allowing users to pass `AbortController.signal`s directly rather than always returning an abort function? I personally find it to be a perfectly reasonable API and it avoids the problem of:
 - having to maintain a reference to the actual promise object returned
 - adding new properties to very well documented native objects

Reply:
Yep, this is perfectly fine to remove, and will probably remove a little of the magic as well.